src/app/TopicDasboard/AddResourceForm.js

```diff
@@ -26,35 +100,48 @@ export class AddResourceForm extends React.Component {
           Add Resource
         </button>
         <form
+          ref={this.Form}
           id="add-resource"
           className="add-resource-form"
           onSubmit={this.handleSubmit}
-          ref={this.Form}
         >
           <div>
             <input type="checkbox" checked={false} readOnly={true} />
             <label htmlFor="add-resource-link" />
             <input
               ref={input => (this.inputUri = input)}
-              type="text"
+              type="url"
               name="add-resource"
+              disabled={this.state.submitting}
               placeholder="http://"
+              onKeyUp={e => {
```

Comment:
Instead of being defined in place, this should probably be lifted up into its own event handler function within the class. You can check the URL there instead of having isUrl be its own method.

Reply:
lifted up as method on the class